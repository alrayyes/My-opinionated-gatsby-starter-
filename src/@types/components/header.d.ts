export interface HeaderInterface {
  siteTitle: string
}
