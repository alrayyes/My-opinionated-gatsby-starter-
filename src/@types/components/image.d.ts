import { FluidObject } from "gatsby-image"

export interface ImageDataInterface {
  placeholderImage: {
    childImageSharp: {
      fluid: FluidObject
    }
  }
}
