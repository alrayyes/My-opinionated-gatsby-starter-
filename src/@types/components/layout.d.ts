export interface LayoutInterface {
  children: React.ReactNode
}

export interface LayoutDataInterface {
  site: {
    siteMetadata: {
      title: string
    }
  }
}
