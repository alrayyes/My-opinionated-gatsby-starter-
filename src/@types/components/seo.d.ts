export interface SEOInterface {
  title: string
  description?: string
  lang?: string
  meta?: Array<
    | {
        property: string
        content: string
      }
    | {
        name: string
        content: string
      }
  >
}

export interface SEODataInterface {
  site: {
    siteMetadata: {
      title: string
      description: string
      author: string
    }
  }
}
