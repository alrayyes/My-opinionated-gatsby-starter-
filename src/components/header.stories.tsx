import { storiesOf } from "@storybook/react"
import React from "react"
import Header from "./header"

storiesOf("Header component", module).add(`default`, () => (
  <Header siteTitle={"My Opinionated Gatsby Default Starter"} />
))
