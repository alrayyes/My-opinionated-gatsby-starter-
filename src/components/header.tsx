import { Link } from "gatsby"
import React from "react"
import { HeaderInterface } from "../@types/components/header"
import styled from "@emotion/styled"
import tw from "twin.macro"

const Header: React.FC<HeaderInterface> = ({ siteTitle }) => {
  const Header = styled.header`
    ${tw`bg-purple-600 mb-6`}
  `

  return (
    <Header>
      <div className={"headerContainer"}>
        <Link className={"headerLink"} to={"/"}>
          {siteTitle}
        </Link>
      </div>
    </Header>
  )
}

export default Header
