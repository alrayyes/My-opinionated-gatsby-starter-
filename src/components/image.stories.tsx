import { storiesOf } from "@storybook/react"
import React from "react"
import Image from "./image"

storiesOf("Image component", module).add(`default`, () => <Image />)
