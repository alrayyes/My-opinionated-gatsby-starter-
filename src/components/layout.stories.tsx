import { storiesOf } from "@storybook/react"
import React from "react"
import Layout from "./layout"

storiesOf("Layout component", module).add(`default`, () => (
  <Layout>content</Layout>
))
