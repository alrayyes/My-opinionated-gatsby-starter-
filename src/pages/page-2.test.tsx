import React from "react"
import renderer from "react-test-renderer"
import * as Gatsby from "gatsby"
import { SEODataInterface } from "../@types/components/seo"
import SecondPage from "./page-2"

const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")

describe("SecondPage", () => {
  beforeEach(() => {
    const data: SEODataInterface = {
      site: {
        siteMetadata: {
          title: "Gatsby Default Starter",
          description:
            "Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.",
          author: "@gatsbyjs",
        },
      },
    }

    useStaticQuery.mockImplementation(() => data)
  })

  it("renders correctly", () => {
    expect(renderer.create(<SecondPage />).toJSON()).toMatchSnapshot()
  })
})
