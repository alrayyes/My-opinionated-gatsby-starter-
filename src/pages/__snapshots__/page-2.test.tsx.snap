// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondPage renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-i2fnqt-Header{--bg-opacity:1;background-color:rgba(128,90,213,var(--bg-opacity));margin-bottom:1.5rem;}",
      }
    }
    data-emotion-css="i2fnqt-Header"
  />,
  <header
    className="css-i2fnqt-Header e1qr1m9j0"
  >
    <div
      className="headerContainer"
    >
      <a
        className="headerLink"
        href="/"
      >
        Gatsby Default Starter
      </a>
    </div>
  </header>,
  <div
    className="layoutContainer"
  >
    <main>
      <h1>
        Hi from the second page
      </h1>
      <p>
        Welcome to page 2
      </p>
      <a
        href="/"
      >
        Go back to the homepage
      </a>
    </main>
    <footer>
      Â© 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
